### PR TITLE
Add a provisioning wrapper for gdeploy

### PR DESCRIPTION
**Pull request by nnDarshan**
This patch adds a provisioning wrapper for gdeploy
which will act as an interface between node_agent and
python-gdeploy modules.

tendrl-bug-id:Tendrl/node-agent#230
Signed-off-by: nnDarshan <darshan.n.2024@gmail.com>

### CODE REVIEW

**Comment by nnDarshan** [comment]
@shtripat @nthomas-redhat  @r0h4n  initial WIP patch for gdeploy wrapper.

File: tendrl/node_agent/provisioner/gluster/provisioner_base.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    def create_gluster_cluster(self, hosts):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def gluster_provision_bricks(self, brick_dictionary, disk_type,
+                                 disk_count, stripe_count):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def gluster_volume_create(self, volume_name, brick_details,
```
Comment (shtripat) [inline tendrl/node_agent/provisioner/gluster/provisioner_base.py]: Do we plan to add more functions like delete_volume, set_volume_option,start/stop_volume etc later here?

Comment (nnDarshan) [inline tendrl/node_agent/provisioner/gluster/provisioner_base.py]: They will be in gdeploy-wrapper of gluster_integration

**Comment by shtripat** [comment]
This could be closed if https://github.com/Tendrl/node-agent/pull/294 is taken in.

**Comment by r0h4n** [comment]
So #294 contains this PR as well? And this can be closed?

**Comment by nnDarshan** [comment]
yes this is not needed as we have https://github.com/Tendrl/node-agent/pull/294

**Pull request closed without merging**